server/utils/auth/command_gate.py

```
@@ -0,0 +1,448 @@
+    safety_code = (
+        "SIGNATURE_MATCHED" if safety_layer == "signature_match"
+        else "SAFETY_BLOCKED" if safety_blocked else None
+    )
+    policy_layer = (
+        "policy_both" if policy_blocked and policy_verdict.deny_rule_id
+            and policy_verdict.allowlist_exhausted
+        else "policy_deny" if policy_blocked and policy_verdict.deny_rule_id
+        else "policy_allow_exhausted" if policy_blocked
+        else None
+    )
+    layers = [l for l in (safety_layer, policy_layer) if l]
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)